src/providers/extraction_providers/extractionProvider.ts

```diff
@@ -0,0 +1,10 @@
+import { Image, ExtractedImage } from '../../models/data_models/image';
+
+export interface ExtractionProvider {
+	/**
+	 * Extracts all labels and objects from an array of input Images
+	 * @param  {Array<Image>} image     Array of images to extract.
+	 * @return {Array<ExtractedImage>}  Array of extractions, where each item pertains to an Image.
+	 */
+	extract(images: Array<Image>): Array<ExtractedImage>;
```

Comment:
In the future we could end up having to extract way more than just images. When that time comes, this interface will probably have to change..

We don't have to worry about that yet though